src/main/java/pl/edu/zut/mad/schedule/search/ScheduleSpecification.java

```diff
@@ -23,9 +25,11 @@ public Predicate toPredicate(Root<Schedule> root, CriteriaQuery<?> query, Criter
         final String searchKey = searchCriteria.getKey();
         final String searchValue = searchCriteria.getValue();
         if (searchKey.equals(DATE_FROM.getKey())) {
-            return builder.greaterThanOrEqualTo(root.get(DATE.getKey()), searchValue);
+            return builder.greaterThanOrEqualTo(root.get(DATE.getKey()), formatDate(searchValue));
         } else if (searchKey.equals(DATE_TO.getKey())) {
-            return builder.lessThanOrEqualTo(root.get(DATE.getKey()), searchValue);
+            return builder.lessThanOrEqualTo(root.get(DATE.getKey()), formatDate(searchValue));
+        } else if (searchKey.equals(DATE.getKey())) {
+            return builder.equal(root.get(DATE.getKey()), formatDate(searchValue));
         } else if (searchKey.equals(FACULTY_ABBREVIATION.getKey())) {
```

Comment:
Do you try to change this massive `else if` to `switch`?

Reply:
Can't be simply converted, because `Fields::getKey()` is not constant. It would require modifying existing `Fields` enum and add method to get enum value by its field.